src/vmm/src/guest_config/x86_64/mod.rs

```diff
@@ -1,17 +1,358 @@
 // Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::collections::HashMap;
+/// Module with CPU templates for x86_64
+pub mod static_cpu_templates;
 
+use std::collections::{HashMap, HashSet};
+
+use kvm_bindings::kvm_msr_entry;
+use static_cpu_templates::*;
+
+use super::cpuid::{CpuidKey, RawCpuid};
+use super::static_templates::{self, msr_entries_to_save};
+use super::templates::x86_64::CpuidRegister;
+use super::templates::{CpuTemplateType, CustomCpuTemplate};
+use crate::arch::x86_64::msr::create_boot_msr_entries;
 use crate::guest_config::cpuid::Cpuid;
+use crate::vstate::vm::Vm;
+
+/// Errors thrown while configuring templates.
+#[derive(Debug, PartialEq, Eq, thiserror::Error)]
+pub enum Error {
+    /// Failure in processing the CPUID in template for x86_64 CPU configuration.
+    #[error(
+        "Template changes a CPUID entry not supported by the host: Leaf: {0:0x}, Subleaf: {1:0x}"
+    )]
+    CpuidFeatureNotSupported(u32, u32),
+    /// Failure in processing the MSRs in template for x86_64 CPU configuration.
+    #[error("Template changes an MSR entry not supported by the host: Register Address: {0:0x}")]
+    MsrNotSupported(u32),
```

Comment:
nit: The base CPUID and MSRs are retrieved from KVM (not from host).
```suggestion
    /// Failure in processing the CPUID in template for x86_64 CPU configuration.
    #[error(
        "Template changes a CPUID entry not supported by KVM: Leaf: {0:0x}, Subleaf: {1:0x}"
    )]
    CpuidFeatureNotSupported(u32, u32),
    /// Failure in processing the MSRs in template for x86_64 CPU configuration.
    #[error("Template changes an MSR entry not supported by KVM: Register Address: {0:0x}")]
    MsrNotSupported(u32),
```

Reply:
fixed